pkg/controller.v1/jax/jaxjob_controller_test.go

```diff
@@ -0,0 +1,320 @@
+				return created.Status.Conditions
+			}, testutil.ConsistentDuration, testutil.Interval).Should(BeComparableTo([]kubeflowv1.JobCondition{
+				{
+					Type:    kubeflowv1.JobCreated,
+					Status:  corev1.ConditionTrue,
+					Reason:  commonutil.NewReason(kubeflowv1.JAXJobKind, commonutil.JobCreatedReason),
+					Message: fmt.Sprintf("JAXJob %s is created.", name),
+				},
+				{
+					Type:    kubeflowv1.JobSuspended,
+					Status:  corev1.ConditionTrue,
+					Reason:  commonutil.NewReason(kubeflowv1.JAXJobKind, commonutil.JobSuspendedReason),
+					Message: fmt.Sprintf("JAXJob %s is suspended.", name),
+				},
+			}, testutil.IgnoreJobConditionsTimes))
+		})
+
+		It("Should delete resources after JAXJob is suspended; Should resume JAXJob after JAXJob is unsuspended", func() {
+			By("By creating a new JAXJob")
+			job.Spec.JAXReplicaSpecs[kubeflowv1.JAXJobReplicaTypeWorker].Replicas = ptr.To[int32](1)
+			Expect(testK8sClient.Create(ctx, job)).Should(Succeed())
+
+			created := &kubeflowv1.JAXJob{}
+			workerPod := &corev1.Pod{}
+			workerSvc := &corev1.Service{}
+
+			// We'll need to retry getting this newly created JAXJob, given that creation may not immediately happen.
+			By("Checking created JAXJob")
+			Eventually(func() bool {
+				err := testK8sClient.Get(ctx, jobKey, created)
+				return err == nil
+			}, testutil.Timeout, testutil.Interval).Should(BeTrue())
+
+			var startTimeBeforeSuspended *metav1.Time
+			Eventually(func() *metav1.Time {
+				Expect(testK8sClient.Get(ctx, jobKey, created)).Should(Succeed())
+				startTimeBeforeSuspended = created.Status.StartTime
+				return startTimeBeforeSuspended
+			}, testutil.Timeout, testutil.Interval).ShouldNot(BeNil())
+
+			By("Checking the created pods and services")
+			Eventually(func() bool {
+				errWorker := testK8sClient.Get(ctx, worker0Key, workerPod)
+				return errWorker == nil
+			}, testutil.Timeout, testutil.Interval).Should(BeTrue())
+			Eventually(func() bool {
+				errWorker := testK8sClient.Get(ctx, worker0Key, workerSvc)
+				return errWorker == nil
+			}, testutil.Timeout, testutil.Interval).Should(BeTrue())
+
+			By("Updating the pod's phase with Running")
+			Eventually(func() error {
+				Expect(testK8sClient.Get(ctx, worker0Key, workerPod)).Should(Succeed())
+				workerPod.Status.Phase = corev1.PodRunning
+				return testK8sClient.Status().Update(ctx, workerPod)
+			}, testutil.Timeout, testutil.Interval).Should(Succeed())
+
+			By("Checking the JAXJob's condition")
+			Eventually(func() []kubeflowv1.JobCondition {
+				Expect(testK8sClient.Get(ctx, jobKey, created)).Should(Succeed())
+				return created.Status.Conditions
+			}, testutil.Timeout, testutil.Interval).Should(BeComparableTo([]kubeflowv1.JobCondition{
+				{
+					Type:    kubeflowv1.JobCreated,
+					Status:  corev1.ConditionTrue,
+					Reason:  commonutil.NewReason(kubeflowv1.JAXJobKind, commonutil.JobCreatedReason),
+					Message: fmt.Sprintf("JAXJob %s is created.", name),
```

Comment:
You can resolve the following errors once you add "namespace/name" here.

https://github.com/kubeflow/training-operator/actions/runs/10708065502/job/29689500512?pr=2194#step:4:174

Reply:
@sandipanpanda We need to address this at all places in this file.